docs/embedded/index.md

```diff
@@ -1,3 +1,155 @@
-# Intorduction
+# Introduction
 
+## Prerequisites
 
+If you did not attend the **Tock Workshop**, please follow the [Setup Tutorial](../tock_workshop/index.md).
+
+## Getting Started
+
+For this track we will be using the **Nucleo-F429ZI** boards. You will need to **change the branch** you are working on, but first make sure you commit your changes.
+
+```shell
+git add .
+git commit -m "tock workshop progress"
+```
+
+Then, to fetch the branch and work on it, run:
+
+```shell
+git fetch
+git checkout track/embedded
+```
+
+The board's main can be found in the `boards/nucleo_f429zi` subfolder. Try to flash the kernel to the board, using the board's `Makefile`. After you are done flashing, connect to the board using `tockloader listen`
+
+```shell
+[INFO   ] No device name specified. Using default name "tock".
+[INFO   ] No serial port with device name "tock" found.
+[INFO   ] Found 2 serial ports.
+Multiple serial port options found. Which would you like to use?
+[0]     /dev/cu.debug-console - n/a
+[1]     /dev/cu.usbmodem1303 - STM32 STLink
+
+Which option? [0] 1
+[INFO   ] Using "/dev/cu.usbmodem1303 - STM32 STLink".
+[INFO   ] Listening for serial output.
+
+tock$
+```
+
+## Customize your kernel
+
+After connecting to Tock's terminal, you can run `help` to see the supported commands. One of them is `reset` and by running it, you can see the default *"welcome"* message.
+
+```shell
+tock$ reset
+Initialization complete. Entering main loop
+tock$
+```
+
+Personalize your kernel, by changing the hostname and the banner.
+
+## Print Counter Capsule
+
+For this task, you will need to build a capsule that prints a custom message each time it receives a print command from an application, along with a message counter representing the number of commands received. Remember that you will need to implement the `SyscallDriver` trait.
+
+### The simple way
+
+Simplest method to do this is to add a `counter` field in the capsule's structure. One issue you will most likely encounter is that the `command` method required by the `SyscallDriver` trait has a immutable reference to `&self`, so you may need to wrap the counter in a wrapper that allows for inner mutability, such as `Cell`.
+
+### The Tock way
+
+One issue with the previous approach is that the counter would be shared between the applications. This could be an issue for mutually distrustful application. Fortunately, Tock has a mechanism in place for such situations, called `Grant`s, which are per-process memory regions allocated in the kernel for a capsule to store that process’s state.
```

Comment:
Maybe we should add a memory diagram to better illustrate where the `Grant`, like this one
<img width="385" height="340" alt="processram" src="https://github.com/user-attachments/assets/a783b4f5-c11b-4224-a96b-7fe8c61cd858" />